enterprise/utils.py

```diff
@@ -1792,17 +1796,32 @@ def customer_admin_enroll_user_with_status(
     succeeded = False
     new_enrollment = False
     enterprise_fulfillment_source_uuid = None
+    emet_enable_auto_upgrade_enrollment_mode = getattr(
+        settings,
+        'ENABLE_ENTERPRISE_BACKEND_EMET_AUTO_UPGRADE_ENROLLMENT_MODE',
+        False,
+    )
     try:
-        # enrolls a user in a course per LMS flow, but this method does not create enterprise records yet so we need to
-        # create it immediately after calling lms_enroll_user_in_course. lms_enroll_user_in_course can return None if
-        # Enrollment already exists, does not fail in this case.
-        new_enrollment = lms_enroll_user_in_course(
-            user.username,
-            course_id,
-            course_mode,
-            enterprise_customer.uuid,
-            is_active=True,
-        )
+        # enrolls a user in a course per LMS flow, but this method doesn't create enterprise records
+        # yet so we need to create it immediately after calling lms_update_or_create_enrollment.
+        # lms_update_or_create_enrollment can return None if Enrollment already exists,
+        # does not fail in this case.
+        if emet_enable_auto_upgrade_enrollment_mode:
+            new_enrollment = lms_update_or_create_enrollment(
+                user.username,
+                course_id,
+                course_mode,
+                True,
```

Comment:
nit: I wonder if it may be slightly more readable to include the kwarg property key when passing `True` so that the code is more self-descriptive/documented (e.g., `is_active=True` below).